server/lib/genome/importers/file_importers/drugbank.rb

```diff
@@ -0,0 +1,246 @@
+
+        when 'external-identifiers'
+            @is_gene_identifier = true
+            @current_target_gene_identifiers = []
+            @current_target_gene_id_resources = []
+
+
+        when 'identifier'
+            if @is_gene_identifier
+                @target_gene_identifier_flag = true
+            end
+
+        when 'resource'
+            if @is_gene_identifier
+                @target_gene_id_resource_flag = true
+            end
+        end
+
+
+
+    end
+
+    def characters(string)
+
+        if @drugbank_id_flag
+            @drugbank_id_flag = false
+            if is_not_salts
+                # p 'DRUGBANK ID: ' + string
+                @current_drug_id = string
+            end
+        end
+
+        if @name_flag
+            if is_not_salts
+                @name_flag = false
+                @is_primary_drug = false
+                # p 'DRUG NAME: ' + string unless string.include?"\n"
+                @current_drug_name = string
+            end
+        end
+
+        if @target_name_flag
+            if @is_target
+                @target_name_flag = false
+                # p 'TARGET NAME: ' + string unless string.include?"\n"
+                @current_target_name = string
+            end
+        end
+
+        if target_action_flag
+            if is_target
+                @target_action_flag = false
+                # p 'ACTION TYPE: ' + string unless string.include?"\n"
+                @current_target_action = string
+            end
+        end
+
+        if target_pubmed_flag
+            if is_target
+                @target_pubmed_flag = false
+                # p 'PUBMED ID: ' + string unless string.include?"\n"
+                @current_target_pmid.append(string) unless string.include?"\n"
+            end
+        end
+
+        if target_gene_identifier_flag
+            if is_gene_identifier
+                @target_gene_identifier_flag = false
+                @current_target_gene_identifiers.append(string) unless string.include?"\n"
+            end
+        end
+
+        if target_gene_id_resource_flag
+            if is_gene_identifier
+                @target_gene_id_resource_flag = false
+                @current_target_gene_id_resources.append(string) unless string.include?"\n"
+            end
+        end
+
+    end
+
+    def end_element(name)
+        case name
+
+        when 'salts'
+            @is_not_salts = true
+
+        when 'external-identifiers'
+            @is_gene_identifier = false
+
+        when 'target'
+            @is_target = false
+            @record.append(current_drug_id,@current_drug_name,@current_target_name,@current_target_action,@current_target_pmid,@current_target_gene_identifiers,@current_target_gene_id_resources)
+            @all_records.append(@record)
+            @record = []
+            #append all sections to a record
+
+        when 'targets'
+        end
+
+    end
+
+    def end_document
+        p 'Finished parsing'
+    end
```

Comment:
I don't believe this method gets called?

Reply:
Oh, I'm guessing this is for Nokigiri? If so maybe just remove the puts statement so it finishes silently